### PR TITLE
Review dialog - open trade

### PR DESCRIPTION
**Pull request by tiltom**

### CODE REVIEW

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 1]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+              isBuy ? styles.orderActionBuy : styles.orderActionSell,
+            )}
+          >
+            {trade?.leverage}x{' '}
+            {t(translations.perpetualPage.reviewTrade.market)}{' '}
+            {getTradePosition(trade?.position)}
+          </div>
+          <div className={styles.orderActionAdditionalInfo}>
+            {toNumberFormat(fromWei(trade?.amount), 3)} BTC @{' '}
+            {isBuy ? '≥' : '≤'} {toNumberFormat(trade.entryPrice, 2)} USD
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:110]: Please use the pair.baseAsset and pair.qouteAsset. If you took this from Trade Form please update it there too.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 2]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+
+        <div className={styles.positionDetailsTitle}>New Position Details</div>
+
+        <div className={styles.positionInfo}>
+          <div className={styles.positionInfoRow}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.reviewTrade.labels.positionSize)}
+            </span>
+            <span className={styles.positionSize}>
+              +{toNumberFormat(fromWei(trade?.amount), 3)} BTC
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:139]: Please use AssetValue with, showPositiveSign

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 3]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+          </div>
+          <div className={styles.orderActionAdditionalInfo}>
+            {toNumberFormat(fromWei(trade?.amount), 3)} BTC @{' '}
+            {isBuy ? '≥' : '≤'} {toNumberFormat(trade.entryPrice, 2)} USD
+          </div>
+        </div>
+
+        <div className={classNames(styles.importantInfo, 'tw-mt-4')}>
+          <div className={styles.tradingFeeWrapper}>
+            <span className="tw-text-gray-10 tw-font-light">
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:116]: Pretty sure this should be font weight regular, also for the following labels.
Please use the values from the design.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 4]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+          <div className={classNames(styles.positionInfoRow, 'tw-mt-2')}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.reviewTrade.labels.margin)}
+            </span>
+            <span>
+              <AssetValue
+                minDecimals={4}
+                maxDecimals={4}
+                mode={AssetValueMode.auto}
+                value={String(margin)}
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:152]: margin is not a wei amount so pass it as number instead.
```suggestion
                value={margin}
```

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 5]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+            </span>
+            <span>{trade.leverage}x</span>
+          </div>
+
+          <div className={classNames(styles.positionInfoRow, 'tw-mt-2')}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.reviewTrade.labels.entryPrice)}
+            </span>
+            <span>
+              {isBuy ? '≥' : '≤'} {toNumberFormat(trade.entryPrice, 2)}
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:170]: Use AssetValue as it should use pair.quoteAsset too.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 6]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+                assetString="BTC"
+              />
+            </span>
+          </div>
+
+          <div className={classNames(styles.positionInfoRow, 'tw-mt-2')}>
+            <span className="tw-text-gray-10 tw-font-light">
+              {t(translations.perpetualPage.reviewTrade.labels.leverage)}
+            </span>
+            <span>{trade.leverage}x</span>
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:162]: Needs to be formatted to a max precision of 2.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 7]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
-      <button onClick={onSubmit}>Open trade</button>
+      <div className={styles.contentWrapper}>
+        <div className={styles.importantInfo}>
+          <div
+            className={classNames(
+              'tw-text-xl tw-font-medium tw-tracking-normal tw-leading-none',
+              isBuy ? styles.orderActionBuy : styles.orderActionSell,
+            )}
+          >
+            {trade?.leverage}x{' '}
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:104]: Format to a precision of 2

**Review by ShynRou — CHANGES_REQUESTED** [review]

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx (hunk 4 above)
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:152]: It's in wei but it's not a string. It's used the same way in `TradeForm`.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx (hunk 3 above)
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:116]: I adjusted font weight for the whole dialog. However, we should finally adjust our code so font weight in design matches font weight in our codebase.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 8]
```diff
@@ -39,15 +61,132 @@ export const TradeReviewDialog: React.FC = () => {
...
+  const margin = useMemo(
+    () =>
+      trade
+        ? getRequiredMarginCollateral(
+            trade!.leverage,
+            marginAccountBalance.fPositionBC,
+            Number(trade!.amount),
+            perpParameters,
+          )
+        : 0,
```
[the remark this answers is not in the document]
Comment (tiltom) [reply src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx]: Updated.

File: src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx [hunk 9]
```diff
@@ -39,15 +71,188 @@ export const TradeReviewDialog: React.FC = () => {
...
+  const margin = useMemo(() => {
+    if (!trade) {
+      return 0;
+    }
+
+    const amount =
+      Number(fromWei(trade.amount)) * getTradeDirection(trade.position);
+
+    return getRequiredMarginCollateral(
+      trade!.leverage,
```
Comment (ShynRou) [inline src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx:83]: Shouldn't be needed here.

```suggestion
      trade.leverage,
```

**Review by ShynRou — APPROVED** [review]

**Pull request merged**